### PR TITLE
Find firstmissingpositiveinteger

### PR DESCRIPTION
**Pull request by Anishluke92**

### CODE REVIEW

File: Find_first_missing_positive_integer.rb [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+In other words, find the lowest positive integer that does not exist in the array. 
+The array can contain duplicates and negative numbers as well.
+
+For example, the input [3, 4, -1, 1] should give 2. The input [1, 2, 0] should give 3.
+
+You can modify the input array in-place.
+
+=end 
+
+def findfirstmissingpostiveint(array)
```
Comment (danielpaul) [inline Find_first_missing_positive_integer.rb]:
```suggestion
def findFirstMissingPositiveInt(array)

```

File: Find_first_missing_positive_integer.rb [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    array = array.sort 
+    array.each_with_index do |number, index|
+        answer = 0
```
Comment (danielpaul) [inline Find_first_missing_positive_integer.rb]: what if the missing number is 0. so [-2, -4, -3, -1, 1, 2]

**Review by danielpaul — CHANGES_REQUESTED** [review]

File: Find_first_missing_positive_integer.rb (hunk 2 above)
Comment (Anishluke92) [reply Find_first_missing_positive_integer.rb]: The input [1, 2, 0] should give 3. Because of this example i had to build code like this.

Comment (Anishluke92) [reply Find_first_missing_positive_integer.rb]: input [3, 4, -1, 1] should give 2.  this too.

Comment (Anishluke92) [reply Find_first_missing_positive_integer.rb]: Technically 0 should come which actually comes. but i had to hard code to move the next number

Comment (Anishluke92) [reply Find_first_missing_positive_integer.rb]: if i remove the condition (answer != 0 ) i can get the correct missing value for all the question except for one " [3, 4, -1, 1] should give 2" but this yeah  [-2, -4, -3, -1, 1, 2] is 0 i can get it.  I dont know which one is needed according to the question the condition is very much needed, else we can avoid the condition and get  the correct and relevant answer to the question

Comment (Anishluke92) [reply Find_first_missing_positive_integer.rb]: I have Modified the coded according to  your requirement need, which satisfices all the conditions.

File: Find_first_missing_positive_integer.rb [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+In other words, find the lowest positive integer that does not exist in the array. 
+The array can contain duplicates and negative numbers as well.
+
+For example, the input [3, 4, -1, 1] should give 2. The input [1, 2, 0] should give 3.
+
+You can modify the input array in-place.
+
+=end 
+
+def FindFirstMissingPostiveInt(array)
```
Comment (danielpaul) [inline Find_first_missing_positive_integer.rb]:
```suggestion
def findFirstMissingPostiveInt(array)

```

File: Find_first_missing_positive_integer.rb [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+    if number == -1 
+      if index == 0 
+        return answer if answer != array[index + 1] && answer != 0
+      else 
+        return answer if answer != array[index + 1]
+      end
+    else 
+      return answer if answer != array[index + 1]
+    end
+  end
```
Comment (danielpaul) [inline Find_first_missing_positive_integer.rb:27]: you can refactor this to be smaller by isolating the common things out

File: Find_first_missing_positive_integer.rb [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+For example, the input [3, 4, -1, 1] should give 2. The input [1, 2, 0] should give 3.
+
+You can modify the input array in-place.
+
+=end 
+
+def FindFirstMissingPostiveInt(array)
+  array = array.sort 
+  array.each_with_index do |number, index|
+    answer = 0
```
Comment (danielpaul) [inline Find_first_missing_positive_integer.rb]:
```suggestion

```
this line is not needed?

**Review by danielpaul — CHANGES_REQUESTED** [review]

**Pull request closed without merging**